qbraid/runtime/result.py

```diff
@@ -139,4 +98,233 @@ def measurement_counts(
+
+        for exp_num, experiment in enumerate(self.result):
+            print(f"Experiment {exp_num} ")
+            print(f"Type: {experiment.result_type}")
+            print(f"Measurements: {experiment.measurements}\n")
+
+    def display_counts(self):
+        """Display the counts of each state in the result along with the type of experiment"""
+
+        for exp_num, experiment in enumerate(self.result):
+            print(f"Experiment {exp_num} ")
+            print(f"Type: {experiment.result_type}")
+            print(f"Counts: {experiment.state_counts}\n")
+
+    def display_metadata(self):
+        """Display the metadata of each experiment in the result"""
+
+        for exp_num, experiment in enumerate(self.result):
+            print(f"Experiment {exp_num} ")
+            print(f"Type: {experiment.result_type}")
+            print(f"Metadata: {experiment.metadata}\n")
+
+    def get_experiment(self, exp_num):
+        """Return the experiment result for a specific experiment number"""
+        if exp_num >= len(self.result):
+            raise ValueError("Experiment number is greater than the number of experiments")
+        return self.result[exp_num]
+
+    def measurements(self) -> Union[np.ndarray, list[np.ndarray]]:
+        """Get the list of measurements from the job result."""
+        measurements = [experiment.measurements for experiment in self.result]
+        if len(measurements) == 1:
+            return measurements[0]
+        return measurements
+
+    def measurement_counts(
+        self, include_zero_values: bool = False, decimal: bool = False
+    ) -> Union[dict[str, int], list[dict[str, int]]]:
+        """Returns the sorted histogram data of the run"""
+        get_counts = [experiment.state_counts for experiment in self.result]
+
+        if len(get_counts) == 0:
+            raise ValueError("No measurements available")
+        batch_counts = [
+            (
+                ResultFormatter.format_counts(counts, include_zero_values=include_zero_values)
+                if experiment.result_type == ExperimentType.GATE_MODEL
+                else counts
+            )
+            for counts, experiment in zip(get_counts, self.result)
+        ]
+        if decimal:
+            decimal_counts = [
+                {int(key, 2): value for key, value in counts.items()} for counts in batch_counts
+            ]
+            return decimal_counts[0] if len(decimal_counts) == 1 else decimal_counts
+
+        if len(batch_counts) == 1:
+            return batch_counts[0]
+
+        return ResultFormatter.normalize_batch_bit_lengths(batch_counts)
+
+    def measurement_probabilities(self, **kwargs) -> dict[str, float]:
+        """Calculate and return the probabilities of each measurement result."""
+        counts = self.measurement_counts(**kwargs)
+        probabilities = ResultFormatter.counts_to_probabilities(counts)
+        return probabilities
```

Comment:
There should be no further structural manipulation of data within the `RuntimeJobResult` class. Instead, job classes should utilize the `ResultFormatter` or similar methods to process the raw results data, such as generating the probability dictionary. These processed results can then be passed and accessed directly through the `RuntimeJobResult`.

Alternatively, you could implement a `@classmethod`, like `RuntimeJobResult.from_dict`, which accepts a dictionary containing raw measurement counts or probabilities, processes them into qBraid-compatible formats, and then creates a new `RuntimeJobResult` with this structured data. For reference, you can look at [qiskit.result.Result.from_dict](https://github.com/Qiskit/qiskit/blob/stable/1.2/qiskit/result/result.py#L112-L131).

Non-structural manipulations, such as formatting or rounding (like you've used in the `measurement_counts` method), *are* acceptable within the `RuntimeJobResult` class. For example, the [qiskit.result.postprocess](https://github.com/Qiskit/qiskit/blob/stable/1.2/qiskit/result/postprocess.py) functions in `qiskit` handle such "post-processing" tasks—these involve aesthetic changes like formatting strings or adjusting precision. 

In summary:

- **Post-processing functions** that modify the formatting of data but return the same data type as the input can remain within the `RuntimeJobResult` class. *These functions should also cache their results to avoid repeated processing.*
- **Data transformations** that create new data types or transform input into different output types (e.g., generating probabilities from measurement counts) should occur before the initialization of `RuntimeJobResult`.